mobile/lib/modules/backup/providers/ios_background_settings.provider.dart

```diff
@@ -0,0 +1,57 @@
+import 'package:hooks_riverpod/hooks_riverpod.dart';
+import 'package:immich_mobile/modules/backup/background_service/background.service.dart';
+
+class IOSBackgroundSettings {
+  final bool appRefreshEnabled;
+  final int numberOfBackgroundTasksQueued;
+  final DateTime? timeOfLastFetch;
+  final DateTime? timeOfLastProcessing;
+
+  IOSBackgroundSettings({
+    required this.appRefreshEnabled,
+    required this.numberOfBackgroundTasksQueued,
+    this.timeOfLastFetch,
+    this.timeOfLastProcessing,
+  });
+}
+
+class IOSBackgroundSettingsState extends StateNotifier<IOSBackgroundSettings?> {
+  final BackgroundService _service;
+  IOSBackgroundSettingsState(this._service) : super(null);
+
+  IOSBackgroundSettings? get settings => state;
+
+  Future<IOSBackgroundSettings> refresh() async {
+    final lastFetchTime = await _service.getIOSBackupLastRun(IosBackgroundTask.fetch);
+    final lastProcessingTime = await _service.getIOSBackupLastRun(IosBackgroundTask.processing);
+    int numberOfProcesses = await _service.getIOSBackupNumberOfProcesses();
+    final appRefreshEnabled = await _service.getIOSBackgroundAppRefreshEnabled();
+
+    // If this is enabled and there are no background processes,
+    // the user just enabled app refresh in Settings.
+    // But we don't have any background services running, since it was disabled
+    // before.
+    if (await _service.isBackgroundBackupEnabled() &&
+        numberOfProcesses == 0) {
+      // We need to restart the background service
+      await _service.enableService();
+      numberOfProcesses = await _service.getIOSBackupNumberOfProcesses();
+    }
+
+    final settings = IOSBackgroundSettings(
+      appRefreshEnabled: appRefreshEnabled,
+      numberOfBackgroundTasksQueued: numberOfProcesses,
+      timeOfLastFetch: lastFetchTime,
+      timeOfLastProcessing: lastProcessingTime,
+    );
+
+    state = settings;
+    return settings;
+  }
+
+}
+
+final iOSBackgroundSettingsProvider = StateProvider<IOSBackgroundSettingsState>(
+  (ref) => IOSBackgroundSettingsState(ref.read(backgroundServiceProvider)),
```

Comment:
I think we should use `ref.watch` here? is there any reason you are choosing to use `ref.read`?

Reply:
Not really sure. I guess watch will update this provider when it changes? This only barely depends on the `backgroundServiceProvider` at all; it just had all the method channel communication functions in it, so I piggybacked there.

I'm still not super familiar with RiverPod and there's likely something I'm not doing correct here. But I made the change and it still seems happily working here.

Reply:
Riverpod suggests using `ref.watch` as much as possible I believe. Only using `ref.read` in button pressing context.

Here is some good info
https://riverpod.dev/docs/concepts/reading#using-ref-to-interact-with-providers

Reply:
Thanks! I gave it a read and it's becoming clearer to me. 

In any event, I changed it to `ref.watch` and looked through the code to find other `ref.read` statements. In this case, the other `ref.read` statements are justified as they are just one-off refreshes (app resume and page load).